index.js

```diff
@@ -5,37 +5,67 @@ const promisify = require('util.promisify');
 
 const promisiedFsRealpath = promisify(fs.realpath);
 
-function realpath(filepath) {
-  if (typeof fs.realpath.native === 'function') {
-    return promisify(fs.realpath.native)(filepath);
-  }
-  const fsBinding = process.binding('fs');
+function isWinRamDiskError(err) {
+  // -4068: EISDIR: illegal operation on a directory, realpath
+  /* Probably RAM-disk on windows.
+     Go straight to the default js
+     implementation.  Otherwise the
+     fsBinding.realpath call may cause
+     the node runtime to abort.
+  */
+  return err.errno === -4068;
+}
 
-  if (fsBinding.realpath) {
-    return new Promise((resolve, reject) => {
-      try {
-        resolve(fsBinding.realpath(filepath, 'utf8'));
-      } catch (e) {
-        reject(e);
-      }
-    });
-  }
+function realpath(filepath, skipToDefault) {
```

Comment:
Could you change this to be named `implementation`or some such so we don't expose the second argument in the API?

`exports` should then be

```js
modules.exports = function realpath(filepath) {
  return implementation(filepath);
};
modules.exports.sync = function realpathSync(filepath) {
  return implementationSync(filepath);
};
```

Reply:
Added wrapper functions to hide the 2nd param